src/char/char.hpp

```diff
@@ -42,6 +42,9 @@ std::chrono::seconds DEFAULT_AUTOSAVE_INTERVAL = 5_min;
 constexpr
 GmLevel default_gm_level = GmLevel::from(0_u32);
 
+// increase the min version when the protocol is incompatible with old m+ versions
+#define MIN_CLIENT_VERSION 1
```

Comment:
Currently manaplus sends version 2 (was sending 1 before implementing the quest log) but version 1 is still compatible thanks to this: https://github.com/themanaworld/tmwa/pull/91/files#diff-49937e616f59c440ac7bc7d0d5ec2237R800

When the client becomes incompatible we simply need to raise MIN_CLIENT_VERSION. Piece of cake.

Usually the version check **should** be handled in login server and not char server but with manaplus it's in char server that the good version is sent because in login server it sends the legacy client version of the *The Mana World* client (20). Manaplus does this to maintain compatibility with very old tmwa and tmwa-based evol. But since it sends the correct version to char it's not a real problem.